README.md

```diff
@@ -998,15 +998,50 @@ should also work if `oomph-lib` has already been configured, using a build direc
+***__Puneet__: Please check: No need to specify the flag during the build and install stages, right? ***
+```bash
+# Configure with range checking 
+cmake -G Ninja -B build -DOOMPH_ENABLE_RANGE_CHECKING=ON
+
+# Build the oomph-lib libraries 
+cmake --build build
+
+# Install
+cmake --install build
+``` 
+The table below provides a complete list of all variables and their default setting, typically chosen such that the fastest possible version of the library is built.
+
+The variables fall into several categories. There are those that get translated into C++ macros, such as 
+* `OOMPH_ENABLE_MPI` which enables `oomph-lib`'s MPI-based parallel capabilities via the C++ macro `OOMPH_HAS_MPI`. It also ensures that the sources code is compiled with a suitable MPI compiler *** __Puneet:__ I think you claimed that his was done automatically. If so, how? If not, does this require the simultaneous specification of mpi compilers for all three languages? **
+* `OOMPH_ENABLE_PARANOID` which enables the compilation of built-in self-tests and diagnostics via the C++ macro `PARANOID`. These tests are very useful during code development but add to the run-time
+* `OOMPH_ENABLE_RANGE_CHECKING`, already discussed above
 
 
+Other variables request that certain third-party libaries are built at the same time as `oomph-lib`. E.g. 
+* `OOMPH_WANT_MUMPS` requests that the parallel frontal solver MUMPS is built
+* `OOMPH_WANT_HYPRE` requests that the Hypre library is built
+* `OOMPH_WANT_TRILINOS` requests that the Trilinos library is built
+* *** __Puneet:__ what else? ***
 
+If any of these are set to `ON`, `CMake` will download the relevant sources and build the libraries as part of the `oomph-lib` installation. It will also (via associated C++ macros, `OOMPH_HAS_MUMPS`, `OOMPH_HAS_HYPRE`, `OOMPH_HAS_TRILINOS`) enable the compilation of certain `oomph-lib` wrappers to these libraries. For instance, if `OOMPH_WANT_MUMPS` is set to `ON` during the project configuration, `oomph-lib`'s C++ sources will be compiled with the C++ macro `OOMPH_HAS_MUMPS` which then enables the creation of a `LinearSolver` object that uses `MUMPS` as the underlying linear solver.
 
+Since third party libraries are not modified/maintained by us, it is, of course, not necessary to rebuild them over and over again. They only have to be installed once in a permanent location, or they may already be installed somewhere on your computer. In this case, their location can be specified via variables such as 
+* `OOMPH_USE_BLAS_FROM` which specifies the absolute path to the directory in which the BLAS library has been installed; this directory should contain an `include` and `lib` subdirectory *** __Puneet:__ Making this up as I go along; we ought to specify what exactly the path is to ***.
```

Comment:
This sounds right (in terms of the path). Modern versions of these libraries will contain something like a `lib/cmake/<BLAS-TYPE>Config.cmake` file that tells CMake everything about how to use the library

Reply:
I'll keep this in there for now (but close the conversation). Just want to make sure that users can also link against old (e.g. oomph-lib autotools based) installations of blas say, which may not have that structure and are given specific instructions what the path should point to.